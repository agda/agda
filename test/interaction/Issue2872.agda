open import Agda.Builtin.List
open import Agda.Builtin.Char

foo : Char → List Char
foo c = c ∷ []

-- Step 1: case split on foo-ls
case-split-test₁ : List Char → List Char
case-split-test₁ ('-' ∷ ls) with (foo '+')
... | foo-ls = {!foo-ls!}
case-split-test₁ ls = ls

-- Step 2: expand ellipsis
case-split-test₂ : List Char → List Char
case-split-test₂ ('-' ∷ ls) with (foo '+')
... | [] = {!.!}
... | x ∷ foo-ls = {!.!}
case-split-test₂ ls = ls

-- WAS: case splitting on foo-ls produces:
--   case-split-test (.'-' ∷ ls) | [] = ?
--   case-split-test (.'-' ∷ ls) | x ∷ foo-ls = ?
-- SHOULD: not put dots in front of literal patterns:
--   case-split-test ('-' ∷ ls) | [] = {!!}
--   case-split-test ('-' ∷ ls) | x ∷ foo-ls = {!!}
