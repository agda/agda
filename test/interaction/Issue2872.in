top_command (cmd_load currentFile [])
goal_command 0 cmd_make_case "foo-ls"
goal_command 1 cmd_make_case "."
goal_command 2 cmd_make_case "."
