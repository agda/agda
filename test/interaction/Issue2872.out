(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : List Char ?1 : List Char ?2 : List Char " nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
((last . 2) . (agda2-make-case-action '("... | [] = ?" "... | x ∷ foo-ls = ?")))
((last . 1) . (agda2-goals-action '(0 1 2)))
((last . 2) . (agda2-make-case-action '("case-split-test₂ ('-' ∷ ls) | [] = ?")))
((last . 1) . (agda2-goals-action '(0 1 2)))
((last . 2) . (agda2-make-case-action '("case-split-test₂ ('-' ∷ ls) | x ∷ foo-ls = ?")))
((last . 1) . (agda2-goals-action '(0 1 2)))
